src/signals/settings/users/containers/Detail/__tests__/Detail.test.js

```diff
@@ -1,39 +1,192 @@
+    useFetchUser.mockImplementationOnce(() => ({ error: { message } }));
+
+    const { getByTestId, getByText } = render(withAppContext(<UserDetail />));
+
+    expect(getByTestId('formAlert')).toBeInTheDocument();
+    expect(getByText(message)).toBeInTheDocument();
+  });
+
+  it('should not patch user data on submit when form data has not been altered', () => {
+    const patch = jest.fn();
+    useFetchUser.mockImplementationOnce(() => ({ data: userJSON, patch }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    fireEvent.click(
+      getByTestId('detailUserForm').querySelector('[type="submit"]')
+    );
+
+    expect(patch).not.toHaveBeenCalled();
+  });
+
+  it('should patch user data on submit', () => {
+    const patch = jest.fn();
+    useFetchUser.mockImplementationOnce(() => ({ data: userJSON, patch }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    fireEvent.change(
+      getByTestId('detailUserForm').querySelector('#last_name'),
+      { target: { value: 'Foo Bar Baz' } }
+    );
+    fireEvent.click(
+      getByTestId('detailUserForm').querySelector('[type="submit"]')
+    );
+
+    expect(patch).toHaveBeenCalled();
+  });
+
+  it('should convert stringified booleans to boolean values', () => {
+    const patch = jest.fn();
+    useFetchUser.mockImplementationOnce(() => ({
+      data: { ...userJSON, is_active: false },
+      patch,
+    }));
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    // ensure that a boolean value is a string value
+    expect(
+      getByTestId('detailUserForm').querySelector(
+        '[name="is_active"][value="false"][checked]'
+      )
+    ).toBeInTheDocument();
+
+    // change a field's value so that the form will be submitted
+    fireEvent.change(
+      getByTestId('detailUserForm').querySelector('#last_name'),
+      { target: { value: 'Foo Bar Baz' } }
+    );
+    fireEvent.click(
+      getByTestId('detailUserForm').querySelector('[type="submit"]')
+    );
+
+    expect(patch).toHaveBeenCalledWith(
+      expect.objectContaining({ is_active: false })
+    );
+  });
+
+  it('should show a notification on successful PATCH', () => {
+    const message = 'Gegevens opgeslagen';
+    useFetchUser.mockImplementationOnce(() => ({ isSuccess: true }));
+
+    const { getByTestId, getByText } = render(withAppContext(<UserDetail />));
+
+    expect(getByTestId('formAlert')).toBeInTheDocument();
+    expect(getByText(message)).toBeInTheDocument();
+  });
+
+  it('should push to history on cancel', () => {
+    useFetchUser.mockImplementationOnce(() => ({ data: userJSON }));
+
+    global.window.confirm = jest.fn();
+
+    const { getByTestId } = render(withAppContext(<UserDetail />));
+
+    fireEvent.click(getByTestId('cancelBtn'));
+
+    expect(global.window.confirm).not.toHaveBeenCalled();
+    expect(push).toHaveBeenCalled();
+  });
+
+  it('should confirm on cancel', () => {
```

Comment:
Don't understand the test. On cancel it should not be confirmed in previous. But in this test it should be confirmed?

Reply:
I'll change the test titles